HOWTO-make-a-release.md

```diff
@@ -0,0 +1,348 @@
+# HOW TO MAKE A RELEASE
+
+This file documents how to make an OpenSSL release.  Please fix any
+errors you find while doing, or just after, your next release!
+
+Releases are done by one person, with a second person acting as the
+reviewer and additional tester.
+
+# Table of contents
+
+-   [Prerequisites](#prerequisites)
+    -   [Check your access](#check-your-access)
+    -   [PGP / GnuPG key](#pgp-gnupg-key)
+    -   [Repositories](#repositories)
+    -   [A method for reviewing](#a-way-to-reviewing)
+-   [Pre-publishing tasks](#pre-publishing-tasks) [once for each version being released]
+    -   [Freeze the source repository](#freeze-the-source-repository) [the day before release]
+    -   [Prepare your repository checkouts](#prepare-your-repository-checkouts)
+    -   [Make sure that the source is up to date](#make-sure-that-the-source-is-up-to-date)
+    -   [Generate the tarball and announcement text](#generating-the-tarball-and-announcement-text) [do not push]
+        -   [OpenSSL 3.0 and on](#openssl-3.0-and-on)
+        -   [OpenSSL before 3.0](#openssl-before-3.0)
+    -   [Update the website locally](#update-the-website-locally) [do not push]
+-   [Publish the release](#publish-the-release)
+-   [Post-publishing tasks](#post-publishing-tasks)
+    -   [Check the website](#check-the-website)
+    -   [Send the announcement mail](#send-the-announcement-mail)
+    -   [Unfreeze the source repository](#unfreeze-the-source-repository)
+    -   [Security fixes](#security-fixes)
+    -   [Keep in touch](#keep-in-touch)
+
+
+# Prerequisites
+
+## Check your access
+
+To perform a release, you must have appropriate access to OpenSSL's
+development host, dev.openssl.org.  To test this, try to log in with
+ssh:
+
+    ssh dev.openssl.org
+
+You must also check that you can perform tasks as the user 'openssl'
+on dev.openssl.org.  When you have successfully logged in, test your
+access to that user with sudo:
+
+    sudo -u openssl id
+
+## PGP / GnuPG key
+
+You must have a PGP / GnuPG key, and its fingerprint should be present
+in the file `doc/fingerprints.txt` in the OpenSSL source.
+
+## Repositories
+
+You must have access to the following repositories:
+
+-   `openssl-git@git.openssl.org:openssl.git`
+
+    This is the usual main source repository
+
+-   `openssl-git@git.openssl.org:openssl-web.git`
+
+    This is the website repository
```

Comment:
Right at the moment this is an OMC only repo. This is a discrepancy if OTC are responsible for doing releases - but don't have all the right access to actually do that.

Reply:
There are a number of such things where we aren't really prepared for a non-OMC person to do a release.  There are a number of files that shouldn't be directly in the web repository (for example newsflash.txt), but rather somewhere else, and have a website rebuild triggered whenever changes to such files are detected.  We already do this for other web pages...

That's off topic for this PR, though

Reply:
Yes - this was more of a side note.